Sources/NIO/ByteBuffer-views.swift

```diff
@@ -96,6 +94,52 @@ public struct ByteBufferView: RandomAccessCollection {
             return ptr.lastIndex(of: element).map { $0 + self._range.lowerBound }
         })
     }
+
+    public func _customContainsEquatableElement(_ element: Element) -> Bool? {
+        return .some(self.withUnsafeBytes { bytes -> Bool in
+            return bytes.contains(element)
+        })
+    }
+
+    public func _copyToContiguousArray() -> ContiguousArray<Element> {
+        return self.withUnsafeBytes { bytes in
+            return bytes._copyToContiguousArray()
+        }
+    }
+
+    public func _copyContents(initializing buffer: UnsafeMutableBufferPointer<UInt8>) -> (Iterator, UnsafeMutableBufferPointer<Element>.Index) {
+        return self.withUnsafeBytes { bytes in
+            let (_, idx) = bytes._copyContents(initializing: buffer)
+            return (IndexingIterator(_elements: self, _position: idx), idx)
+        }
+    }
+
+    public func _failEarlyRangeCheck(_ index: Index, bounds: Range<Index>) {
+        precondition(
+            self.startIndex <= bounds.lowerBound && bounds.upperBound <= self.endIndex,
+            "Bounds out of buffer's range")
```

Comment:
I don't believe this precondition is necessary: I think Swift guarantees that we shouldn't need it. @lorentey, do we?

Reply:
@weissi I added this precondition as you suggested. But if it's not necessary, I think we don't need to implement `_failEarlyRangeCheck` methods because `index` and `range` parameter are already checked for lying within `bounds` in `Collection`. How do you think?